tests/testthat/test-facets.R

```diff
@@ -47,3 +47,49 @@ test_that("3 facets becomes 3 panels", {
   u <- unique(trace.axes.df)
   expect_identical(nrow(u), 3L)
 })
+
+# expect a certain number of _unique_ [x/y] axes
+expect_axes <- function(info, n, axis = "x") {
+  pattern <- paste0("^", axis, "axis([0-9]+)?$")
+  #n.axes <- length(grep(pattern, names(info$kwargs$layout)))
+  axes <- with(info$kwargs, layout[grepl(pattern, names(layout))])
+  n.axes <- length(axes)
+  ranges <- do.call("rbind", lapply(axes, function(x) x$range))
+  expect_identical(nrow(unique(ranges)), as.integer(n))
+}
+
+no_panels <- ggplot(mtcars, aes(mpg, wt)) + geom_point()
+
+test_that("facet_wrap(..., scales = 'free') creates interior scales", {
+  free_both <- no_panels + facet_wrap(~am+vs, scales = "free")
```

Comment:
We want each of these new plots to be a part of the visual, image diff test suite.
Please add `save_outputs(free_both, "both_scales_free")`.